sharing/src/main/scala/io/delta/sharing/spark/DeltaSharingDataSource.scala

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.sharing.spark
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql.delta.{
+  DeltaColumnMapping,
+  DeltaErrors,
+  DeltaTableUtils => TahoeDeltaTableUtils
+}
+import org.apache.spark.sql.delta.commands.cdc.CDCReader
+import org.apache.spark.sql.delta.metering.DeltaLogging
+import org.apache.spark.sql.delta.schema.SchemaUtils
+import org.apache.spark.sql.delta.sources.{DeltaDataSource, DeltaSQLConf}
+import io.delta.sharing.client.{DeltaSharingClient, DeltaSharingRestClient}
+import io.delta.sharing.client.model.{Table => DeltaSharingTable}
+import io.delta.sharing.client.util.{ConfUtils, JsonUtils}
+import org.apache.hadoop.fs.Path
+
+import org.apache.spark.SparkEnv
+import org.apache.spark.delta.sharing.PreSignedUrlCache
+import org.apache.spark.sql.{SparkSession, SQLContext}
+import org.apache.spark.sql.execution.datasources.HadoopFsRelation
+import org.apache.spark.sql.execution.streaming.Source
+import org.apache.spark.sql.sources.{
+  BaseRelation,
+  DataSourceRegister,
+  RelationProvider,
+  StreamSourceProvider
+}
+import org.apache.spark.sql.types.StructType
+
+/**
+ * A DataSource for Delta Sharing, used to support all queries on a delta sharing table.
+ */
```

Comment:
what are the `all queries`? Please add the details: time travel, with filter, with limit etc?

Reply:
updated